aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java

```diff
@@ -257,8 +258,11 @@ synchronized void releaseSubscription(String subscriptionId) throws ApiException
             subscription.awaitSubscriptionCompleted();
         }
 
+        subscriptions.remove(subscriptionId);
```

Comment:
Previously there was no logic for cleaning up the subscriptions array.  Adding this here ensures that:
 - the subscription objects are properly removed from `subscriptions`
 - the websocket is closed when `subscriptions.size() == 0`.